includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1289,6 +1289,7 @@ public function process_payment( $order_id ) {
 				$order->add_order_note( $note );
 			}
 
+			add_filter( 'woocommerce_return_previous_exceptions', '__return_true' );
```

Comment:
nit: Let's add a comment why this is necessary.

Reply:
Also do you think we should remove the filter somewhere after the exception is thrown? I am guessing that's not necessary as the request ends after the exception.

Reply:
Comment added [5b4a8b3](https://github.com/Automattic/woocommerce-payments/pull/8920/commits/5b4a8b31c7b6ee1ad6385fa9cc4db5116cb39756)

Reply:
>Also do you think we should remove the filter somewhere after the exception is thrown? 

We should remove the filter but after the Exception is thrown, we can't remove it.  

>I am guessing that's not necessary as the request ends after the exception.

That's what I think.